### ISSUE TITLE
IllegalArgumentException: pointerIndex out of range

### ISSUE THREAD

**Issue by 73k05** [Turtlebody/android-image-preview#8]
Hi @wangsun6 thank you for this wonderful open source project!

I found a new crash on the lib.

This could be reproduced if you click/touch many times with two hands and quickly on the picture after you have selected one and it is displayed with the mini thumbnail...

Enclosed the stack
```
2020-07-17 15:19:21.486 27796-27796/com.greentoad.turtlebody.imagepreview.sample E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.greentoad.turtlebody.imagepreview.sample, PID: 27796
    java.lang.IllegalArgumentException: pointerIndex out of range
        at android.view.MotionEvent.nativeGetAxisValue(Native Method)
        at android.view.MotionEvent.getX(MotionEvent.java:2367)
        at androidx.viewpager.widget.ViewPager.onInterceptTouchEvent(ViewPager.java:2072)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2609)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:465)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1849)
        at android.app.Activity.dispatchTouchEvent(Activity.java:3993)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:423)
        at android.view.View.dispatchPointerEvent(View.java:13675)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5483)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5286)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4789)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4842)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4808)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4948)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4816)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5005)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4789)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4842)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4808)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4816)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4789)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7512)
        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7481)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7442)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7637)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:189)
        at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
        at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:179)
        at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:7588)
        at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:7661)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1041)
        at android.view.Choreographer.doCallbacks(Choreographer.java:864)
        at android.view.Choreographer.doFrame(Choreographer.java:792)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1026)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7397)
2020-07-17 15:19:21.486 27796-27796/com.greentoad.turtlebody.imagepreview.sample E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)
```

**Comment by wangsun6** [comment]
Thanks for report...

We have sample application released in play store..

Please install the app from play store and let us know if you get the
crash..

Play store link is in github page.

**Comment by 73k05** [comment]
Hey, thank you for the quick reply, yes I've got the exacte same crash from the app I just downloaded from the playstore v1.0.2 on playstore. Last updated code on master. I've found this https://stackoverflow.com/questions/16459196/java-lang-illegalargumentexception-pointerindex-out-of-range-exception-dispat but I don't know where to put it yet.

**Comment by wangsun6** [comment]
The crash is related to the library so you can't do anything.

Also I can't produce the error. Can u do small screen record and send me
the video, please?

**Comment by 73k05** [comment]
I just sent it to your email address, let me know if I can help.

**Comment by wangsun6** [comment]
I can't produce the error. But i will update the code 2morrow.

**Comment by 73k05** [comment]
I can try to do it as I can reproduce, I think I have to create a new class extending the ViewPager `preview_fragment_viewpager` in `ImagePreview.kt` and override `onTouchEvent` & `onInterceptTouchEvent` and surround the `super` call by try/catch

**Comment by wangsun6** [comment]
Try..and let me know if you get any solution.

**Comment by 73k05** [comment]
I think I fixed it in PR #9 , the crash does not happen anymore and I get the following error logs:

![Screenshot 2020-07-17 at 23 32 02](https://user-images.githubusercontent.com/9129812/87832202-bd476600-c885-11ea-8a98-48712a3064f8.png)

Let me know if it's ok for you as you review the code.

**Comment by 73k05** [comment]
PR merged
